custom_components/tuya_local/devices/djive_arc_fan.yaml

```diff
@@ -0,0 +1,101 @@
+name: djive Flowmate ARC Humidifier
+products:
+  - id: bf5fd02f13d3f6d92exy2t
+    name: djive ARC Humidifier
+primary_entity:
+  entity: fan
+  name: Fan
+  dps:
+    - id: 1
+      name: switch
+      type: boolean
+    - id: 2
+      name: preset_mode
+      type: string
+      optional: true
+      mapping:
+        - dps_val: close
+          value: standard
+        - dps_val: auto
+          value: auto
+        - dps_val: strong
+          value: max
+        - dps_val: fresh
+          value: fresh
+        - dps_val: sleep
+          value: sleep
+    - id: 3
+      name: speed
+      type: integer
+      range:
+        min: 0
+        max: 9
+      mapping:
+        - scale: 0.09
+    - id: 5
+      type: boolean
+      name: oscillate
+secondary_entities:
+  - entity: select
+    category: config
+    name: Humidifier
+    icon: "mdi:air-humidifier"
+    dps:
+      - id: 105
+        name: option
+        type: string
+        mapping:
+          - dps_val: close
+            value: "off"
+          - dps_val: 1
+            value: low
+          - dps_val: 2
+            value: medium
+          - dps_val: 3
+            value: high
+          - dps_val: auto
+            value: auto
+  - entity: binary_sensor
+    name: Fault
+    class: problem
+    category: diagnostic
+    dps:
+      - id: 24
```

Comment:
Yeah agreed, but I wasn't able to figure out what the actual fault codes mean / or what fault codes are possible. But I can also just add this to the primary entity:
```
    - id: 24
      type: bitfield
      name: fault_code
```
Is this what you mean?

Reply:
Yes, that was what I meant by "raw" - the original integer value without any attempt to turn them into meaningful fault codes.  It will help with figuring out the code meanings so you can add more binary sensors for specific codes later. Tank empty (or low) is probably one that would be useful to separate out if you can identify it.

Reply:
Tank empty is handled by a different code, but the code never gets reported by the device. If I figure out what the fault codes mean, I will update the config accordingly.